[FEAT]: New event subscription in Extra

Doominika opened the pull request:
## Description
Implement new registration for events based on new event subscription API.

djenczewski committed:
feat(privmx-endpoint-extra): first implementation of new subscription methods

Doominika committed:
refactor: update EventType class

Doominika committed:
refactor: add internal isLibEvent function

Doominika committed:
refactor: add new enum with modules (contains core)

Doominika committed:
fix: use new eventName EventType param

Doominika committed:
docs: update and add missing documentation

Doominika committed:
refactor: update unregisterCallbacks method to accept many callbacks

Doominika committed:
refactor: change type of event result class for StoreFileUpdatedEvent

djenczewski commented on privmx-endpoint-extra/src/commonMain/kotlin/com/simplito/kotlin/privmx_endpoint_extra/lib/PrivmxEndpoint.kt:
```diff
@@ -48,47 +64,47 @@ constructor(
 
     /**
-     * Unregisters all callbacks registered by [registerCallback] and identified with given Context.
+     * Unregisters all callbacks identified by given group reference.
```
maybe it would be better to use the plural?

djenczewski committed:
chore(privmx-endpoint-extra): make fixes in PR

djenczewski committed:
Sync refactor/implement-new-subscription-methods-extra with feat/implement-2.6

# Conflicts:
#	privmx-endpoint-extra/src/commonMain/kotlin/com/simplito/kotlin/privmx_endpoint_extra/events/EventType.kt

djenczewski committed:
fix(privmx-endpoint-extra): compilator warning about exposed internal parameter

Doominika committed:
refactor: rename event type for kvdb stats-changed event

Doominika committed:
docs: change unregisterCallbacks method description

Doominika committed:
docs: change unregisterCallbacks method description

filipg43 commented on privmx-endpoint-extra/src/commonMain/kotlin/com/simplito/kotlin/privmx_endpoint_extra/lib/PrivmxEndpoint.kt:
```diff
@@ -107,158 +123,241 @@ constructor(
+)
+
+/**
+ * An result for single [CallbackRegistration].
```
A result

filipg43 commented on privmx-endpoint-extra/src/commonMain/kotlin/com/simplito/kotlin/privmx_endpoint_extra/events/EventDispatcher.kt:
```diff
@@ -85,46 +90,176 @@ class EventDispatcher(
+        }
+
+    /**
+     * Get reference to list for adding or removing callbacks.
```
Get reference to a list that can be used to add or remove callbacks.

Doominika committed:
docs: fix descriptions
